scipy/special/_amos.c

```diff
@@ -0,0 +1,6206 @@
+        return nz;
+    }
+
+    tol = fmax(d1mach[3], 1e-18);
+    k1 = i1mach[14];
+    k2 = i1mach[15];
+    r1m5 = d1mach[4];
+    k = ( abs(k1) > abs(k2) ? abs(k2) : abs(k1) );
+    //
+    // ELIM IS THE APPROXIMATE EXPONENTIAL UNDER- AND OVERFLOW LIMIT
+    //
+    elim = 2.303 * (k*r1m5 - 3.0);
+    exr = cos(xx);
+    exi = sin(xx);
+    ey = 0.0;
+    tay = fabs(yy + yy);
+    if (tay < elim) { ey = exp(-tay); }
+    if (yy < 0.0) {
+        /* 90 */
+        c1 = CMPLX(exr, exi);
+        c2 = ey*CMPLX(exr, -exi);
+    } else {
+        c1 = ey*CMPLX(exr, exi);
+        c2 = CMPLX(exr, -exi);
+    }
+
+    nz = 0;
+    rtol = 1.0 / tol;
+    ascle = 1e3*d1mach[0]*rtol;
+    for (i = 1; i< (n+1); i++)
+    {
+        aa = creal(cwrk[i-1]);
+        bb = cimag(cwrk[i-1]);
+        atol = 1.0;
+        if (fmax(fabs(aa), fabs(bb)) <= ascle) {
+            aa *= rtol;
+            bb *= rtol;
+            atol = tol;
+        }
+
+        st = CMPLX(aa, bb) * c2 * atol;
+        aa = creal(cy[i-1]);
+        bb = cimag(cy[i-1]);
+        atol = 1.0;
+        if (fmax(fabs(aa), fabs(bb)) <= ascle) {
+            aa *= rtol;
+            bb *= rtol;
+            atol = tol;
+        }
+
+        st -= CMPLX(aa, bb) * c1 * atol;
+        cy[i-1] = st*hci;
+        if ((st == 0.0) && (ey == 0.0)) { nz += 1; }
+    }
+    return nz;
+}
+
+
+int amos_binu(
+    double complex z,
+    double fnu,
+    int kode,
+    int n,
+    double complex *cy,
+    double rl,
+    double fnul,
+    double tol,
+    double elim,
+    double alim
+) {
+
+    //***BEGIN PROLOGUE  ZBINU
+    //***REFER TO  ZBESH,ZBESI,ZBESJ,ZBESK,ZAIRY,ZBIRY
+    //
+    //     ZBINU COMPUTES THE I FUNCTION IN THE RIGHT HALF Z PLANE
+    //
+    //***ROUTINES CALLED  AZABS,ZASYI,ZBUNI,ZMLRI,ZSERI,ZUOIK,ZWRSK
+    //***END PROLOGUE  ZBINU
+
+    double complex cw[2] = { 0. };
+    double az, dfnu;
+    int inw, nlast, nn, nui, nw, nz;
+
+    nz = 0;
+    az = cabs(z);
+    nn = n;
+    dfnu = fnu + n - 1;
+    if ((az <= 2.) || (az*az*0.25 <= (dfnu + 1.0))) {
+        /* GOTO 10 */
+        nw = amos_seri(z,fnu, kode, n, cy, tol, elim, alim);
+        inw = abs(nw);
+        nz += inw;
+        nn -= inw;
+        if (nn == 0) { return nz; }
+        if (nw >= 0) { return nz; }
+        dfnu = fnu + nn - 1;
+    }
+    /* GOTO 30 conditions*/
+    //
+    // ASYMPTOTIC EXPANSION FOR LARGE Z
+    //
+    if (az >= rl) {
+        if ((dfnu <= 1.0) || ((dfnu > 1.0) && (az+az >= dfnu*dfnu))) {
+            //
+            // MILLER ALGORITHM NORMALIZED BY THE SERIES
+            //
+            nw = amos_asyi(z, fnu, kode, n, cy, rl, tol, elim, alim);
+            if (nw < 0) {
+                nz = -1;
+                if (nw == -2) {
+                    nz = -2;
+                }
+                return nz;
+            }
+            return nz;
+        }
+    }
+    /* 40 */
+    if ((az < rl) || (az+az >= dfnu*dfnu)) {
+        nw = amos_mlri(z, fnu, kode, n, cy, tol);
```

Comment:
```suggestion
        //
        // MILLER ALGORITHM NORMALIZED BY THE SERIES
        //
        nw = amos_mlri(z, fnu, kode, n, cy, tol);
```

This is where this comment should go.